R/create_target_metadata_item.R

```diff
@@ -149,6 +151,20 @@ check_target_keys <- function(target_keys, call = rlang::caller_env()) {
       call = call
     )
   }
+  is_gte_v4_0_1 <- hubUtils::version_gte(
+    "v4.0.1",
+    schema_version = schema_version
+  )
+  target_key_n <- length(target_keys)
+  if (is_gte_v4_0_1 && target_key_n > 1L) {
+    cli::cli_abort(
+      c(
+        "!" = "{.arg target_keys} must be a named {.cls list} of
+        length {.val {1L}} not {.val {target_key_n}}."
+      ),
+      call = call
+    )
+  }
```

Comment:
I restricted this to the relevant versions to ensure it matches what's actually described in earlier schema